lesson_7/edmond_sabou/lesson7_2.py

```diff
@@ -0,0 +1,26 @@
+def safe_divide(func):
+
+    def verify(*args):
+
+        try:
+            result = func(*args)
+        except ZeroDivisionError as zero_err:
+            print(f"Sorry! {zero_err} occurred. Try NOT dividing by zero!")
+            pass
```

Comment:
A better solution here would be to raise a new Exception. You could use the following:
```
raise Exception(f"Sorry! {zero_err} occurred. Try NOT dividing by zero!")
```
And remove the prints and pass

Reply:
made the change to raise Exception. I think I understand why this is better: because of the exit code, which is now 1, signaling that an actual error was encountered. until now even if an error has occurred the 0 exit code was printed.